integration/tests/cook/test_basic.py

```diff
@@ -211,11 +212,19 @@ def test_mea_culpa_retries(self):
 
     def test_compute_cluster(self):
         settings_dict = util.settings(self.cook_url)
-        # For now, we only have one compute cluster. This could be wrong with future
-        # refactors, but for now, this is the only compute cluster we have, so we should
-        # expect it.
         expected_compute_cluster = settings_dict['compute-clusters'][0]['config']['compute-cluster-name']
-        expected_mesos_framework = settings_dict['compute-clusters'][0]['config']['framework-id']
+        compute_cluster_factory_fn = settings_dict['compute-clusters'][0]['factory-fn']
+        if compute_cluster_factory_fn == 'cook.kubernetes.compute-cluster/factory-fn':
+            expected_compute_cluster_type = 'kubernetes'
+        elif compute_cluster_factory_fn == 'cook.mesos.mesos-compute-cluster/factory-fn':
+            expected_compute_cluster_type = 'mesos'
+        else:
+            expected_compute_cluster_type = 'UNKNOWN - TEST WILL FAIL'
+        expected_compute_type = settings_dict['compute-clusters'][0]['config']['compute-cluster-name']
+        if 'framework-id' in settings_dict['compute-clusters'][0]['config']:
```

Comment:
Why not use 
     settings_dict['compute-clusters'][0]['config'].get('framework-id',None)

Reply:
Done